src/main/java/seedu/address/model/ItemModelManager.java

```diff
@@ -523,6 +514,58 @@ private VisualizeList getNextTask() {
         return result;
     }
 
+    /**
+     * Turns off the priority mode.
+     */
+    private void toggleOffPriorityMode() {
+        if (timer != null) {
+            timer.cancel();
+            timer = null;
+        }
+
+        this.sortedTask = null;
+        if (visualList instanceof TaskList) {
+            Platform.runLater(new Runnable() {
+                @Override
+                public void run() {
+                    visualList.clear();
+                    for (Item item : taskList) {
+                        visualList.add(item);
+                    }
+                }
+            });
+        }
+        this.priorityMode = false;
```

Comment:
Good work! But perhaps you can consider creating another class to handle the toggling on and off of priority mode? This can help to segregate the functionalities of the priority mode from the model itself.

Reply:
The reason why the toggling of priority mode is done in the ```ItemModel``` is because there is a need to maintain the timer within the ```ItemModel``` and to destroy it when the priority mode is deactivated. I will explore ways to remove it from the ```ItemModel``` in future iteration 👍.